primitives/npos-elections/solution-type/src/lib.rs

```diff
@@ -0,0 +1,243 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Proc macro for a npos solution type.
+
+use proc_macro::TokenStream;
+use proc_macro2::{Ident, Span, TokenStream as TokenStream2};
+use proc_macro_crate::{crate_name, FoundCrate};
+use quote::quote;
+use syn::parse::{Parse, ParseStream, Result};
+
+mod codec;
+mod from_assignment_helpers;
+mod index_assignment;
+mod single_page;
```

Comment:
I feel like the module name doesn't make much sense anymore.

Reply:
soon I will introduce `paged` into the jargon of this and other crates, I think it is okay.